### PR TITLE
adds search box to channels modal

### PR DESCRIPTION
**Pull request by mugiwarafx**

### CODE REVIEW

**Review by darknao — COMMENTED** [review]
First, please rebase your branch. There is a lot of commit unrelated with this PR right now, a rebase will fix that.

I think it would be nice if the search box was getting the focus when the modal is opened.

File: src/mote/static/jscn/fragment.js [hunk 1]
```diff
@@ -123,7 +124,37 @@ async function populate_channel_list() {
...
+        );
+        $("#searchInput").on("keyup", function () {
+            var value = $(this).val().toLowerCase();
+            $("#listchan-uols *").filter(function () {
+                $(this).toggle($(this).text().toLowerCase().indexOf(value) > -1);
+          });
+        });
+  } else {
+      $("#searchInput").val("");
+  }
```
[the remark this answers is not in the document]
Comment (mugiwarafx) [reply src/mote/static/jscn/fragment.js:157]: Hi @darknao, I will do that!

In my last commit I modified this code block slightly, this way it searches/filters using only the channel's name, also it fix a bug who preveneted "source span" from render once you start typing a search.

```js
if (!$("#searchInput").length > 0) {
    channelHead.parentElement.parentElement.insertAdjacentHTML(
        "afterend",
        `<input id="searchInput" class="form-control mb-3" type="text" placeholder="Search channel... 🔎️">`
    );
    $("#searchInput").on("keyup", function () {
        var value = $(this).val().toLowerCase();
        $("#listchan-uols .head").filter(function() {
        $(this)
          .parent()
          .toggle($(this).text().toLowerCase().indexOf(value) > -1);
        });
    });
} else {
    $("#searchInput").val("");
}
```

Let me rebase, add focus and I will ask for a revision again. :)

[the remark this answers is not in the document]
Comment (darknao) [reply src/mote/static/jscn/fragment.js:160]: I mean, this block is currently here 2 times in the same file.

[the remark this answers is not in the document]
Comment (mugiwarafx) [reply src/mote/static/jscn/fragment.js:160]: OK... time to clean my eyeglasses! 😂️

I fix it in a sec 🙂️

**Comment by darknao** [comment]
Ok, one last thing : 
This is how you should rebase your work, in order to keep it in sync with the upstream repository : 
First, configure the upstream remote : 
```
git remote add upstream https://github.com/fedora-infra/mote.git
```
Fetch all updates from it
```
git fetch upstream
```
Then, rebase your current branch
```
git rebase upstream/main
```

At this point, you should have all the recent upstream changes + your changes on top of it.
Review your code, just in case, and if everything is ok, do a force push.
You can also take this opportunity to squash your commits to make the git history even cleaner :)

**Comment by mugiwarafx** [comment]
many thanks @darknao for your explanations, I really appreciate it! :pray: 

I hope I followed your instructions properly, otherwise just let me know :)

If there is anything left to do, I am a listening port 🤓️

**Review by darknao — APPROVED** [review]

**Comment by darknao** [comment]
Looks perfect :) 
Good job and thanks for your contribution :+1:

**Review by t0xic0der — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**